Lib/test/test_dataclasses.py

```diff
@@ -3169,6 +3169,37 @@ def __post_init__(self, y):
             replace(c, x=3)
         c = replace(c, x=3, y=5)
         self.assertEqual(c.x, 15)
+
+    def test_recursive_repr(self):
+        @dataclass
+        class C:
+            f:"C"
```

Comment:
Needs a space after the colon to be consistent with the rest of this module. This applies throughout the new tests.